test/unit/ios-log-specs.js

```diff
@@ -55,4 +55,45 @@ describe('system logs', () => {
 
     await log.stopCapture();
   });
+
+  it('should rotate log buffer', async function () {
+    const maxBufferSize = 10;
+    const sliceSizeLimit = maxBufferSize / 2;
+    sliceSizeLimit.should.be.below(maxBufferSize);
+    const logRecordsCount = maxBufferSize * 2;
+    logRecordsCount.should.be.above(maxBufferSize);
+
+    let log = new IOSLog({sim, showLogs: false});
+    log.maxBufferSize = maxBufferSize;
+    log.logIdxSinceLastRequest.should.be.below(0);
+    await log.startCapture();
+    let recentLogs = await log.getLogs();
+    recentLogs.should.have.lengthOf(0);
+    log.logIdxSinceLastRequest.should.be.below(0);
+
+    for (let i = 1; i <= logRecordsCount; ++i) {
+      await fs.writeFile(tmpSystemLog, `${i}\n`, {flag: 'a'});
+      // This is to make sure the new entry has been captured on slow Travis nodes
+      await B.delay(10);
+      if (i >= sliceSizeLimit && i % sliceSizeLimit === 0) {
+        // on some slow system (e.g., Travis) need a moment
+        let previousRecentLogs = recentLogs;
+        recentLogs = await log.getLogs();
+        if (previousRecentLogs.length && recentLogs.length) {
+          previousRecentLogs[0].message.should.not.be.equal(recentLogs[0].message);
+        }
+        recentLogs.should.have.length.within(1, sliceSizeLimit);
+        log.logIdxSinceLastRequest.should.be.within(0, log.logs.length);
+      }
+      log.logs.should.have.length.at.most(maxBufferSize);
+    }
+    log.logs.should.have.length.within(2, maxBufferSize);
+
+    const firstBufferMessage = parseInt(log.logs[0].message, 10);
+    firstBufferMessage.should.be.within(1, logRecordsCount - log.logs.length);
```

Comment:
this happens after I clone an appium module and then installing missing npm modules for gulp to be able to run "gulp once"